### PR TITLE
interactive bots: create salesforce bot.

### PR DESCRIPTION
**Pull request by fredfishgames**
This is a salesforce bot.
I haven't done mypy because it came up with errors which I couldn't fix.
It supports getting records from any salesforce database, and lets users easily add their own commands.
You can also post links to it, and it will reply with details of that link.

### CODE REVIEW

File: zulip_bots/zulip_bots/bots/salesforce/doc.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Salesforce bot
+
+The salesforce bot can get records from your salesforce database.
+It can also show details about any saleforce links that you post.
+
```
Comment (eeshangarg) [inline zulip_bots/zulip_bots/bots/salesforce/doc.md]: Capitalize all occurrences of `salesforce` :)

Comment (eeshangarg) [inline zulip_bots/zulip_bots/bots/salesforce/doc.md]: Capitalise all occurrences of `salesforce` :)

File: zulip_bots/zulip_bots/bots/salesforce/doc.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+## Setup
+
+1. Create a user in saleforce that the bot can use to access salesforce.
+Make sure it has the appropriate permissions to access records.
+2. In `salesforce.conf` paste the saleforce `username`, `password` and
+`security_token`.
+3. Run the bot as explained [here](https://zulipchat.com/api/running-bots#running-a-bot)
+
```
Comment (eeshangarg) [inline zulip_bots/zulip_bots/bots/salesforce/doc.md]: Capitalise all occurrences of `salesforce` :)

File: zulip_bots/zulip_bots/bots/salesforce/doc.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+object | [x] | str | The salesforce object type in `object_types` | `None`
+query | [ ] | str | The SOQL query to access this object* | `'SELECT {} FROM {} WHERE Name LIKE %\'{}\'% LIMIT {}'`
+description | [x] | str | What does the command do? | `None`
+template | [x] | str | Example of the command | `None`
+rank_output | [ ] | boolean | Should the output be ranked? (1., 2., 3. etc.) | `False`
+force_keys | [ ] | list[str] | Values which should always be shown in the output | `[]`
+callback | [ ] | callable** | Custom handling behaviour | `None`
+
+
+
```
Comment (eeshangarg) [inline zulip_bots/zulip_bots/bots/salesforce/doc.md]: While we are here, could you please remove the superfluous newlines here? :)

**Comment by fredfishgames** [comment]
This the the only error I get when running mypy:
```
zulip_bots/zulip_bots/bots/salesforce/test_salesforce.py:162: error: No overload variant of "assertRaises" of "TestCase" matches argument types [builtins.Exception]
```

**Comment by fredfishgames** [comment]
It doesn't make sense though, as assertRaises is supposed to have one parameter, which is an exception

**Comment by fredfishgames** [comment]
I couldn't find out how to fix that error, so instead of looking for the program being quit, it looks for the error message being shown.

**Comment by fredfishgames** [comment]
I messed up the PR, here is a new one: #278

**Pull request closed without merging**